### PR TITLE
Add blacklist behavior

### LINKED ISSUE #19: Don't do anything on XmlHttpRequest
We need a way to blacklist path, url, methods...

I have some Ajax calls returning JSON in my app, and it should not fetch Seo.

BUT maybe some people load full pages in Ajax... So maybe we should check the request / response format ?

### PR DESCRIPTION
**Pull request by pyrech**
Fixes #19

### CODE REVIEW

File: src/Bridge/Symfony/DataCollector/SeoManager.php [hunk 1]
```diff
@@ -94,6 +96,18 @@ public function fetch(Fetcher $fetcher, string $path, string $domainAlias = null
     /**
      * {@inheritdoc}
      */
+    public function overrideHtml(string $html): string
+    {
+        if ($this->data['status'] === SeoOverrideDataCollector::STATUS_NOT_RUN) {
+            $this->data['status'] = SeoOverrideDataCollector::STATUS_BLACKLISTED;
```
Comment (damienalexandre) [inline src/Bridge/Symfony/DataCollector/SeoManager.php]: :+1:

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 ## Not yet released
 
+* Remove SeoManagerInterface
```
Comment (pyrech) [inline CHANGELOG.md]: Add the mention `[BC BREAK]`

**Comment by damienalexandre** [comment]
Looks like it needs a rebase :)

**Comment by pyrech** [comment]
Last call for review before merge :slightly_smiling_face:

File: src/Bridge/Symfony/Blacklister/Not2xxBlacklister.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+use Joli\SeoOverride\Bridge\Symfony\Blacklister;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class Not2xxBlacklister implements Blacklister
+{
+    public function isBlacklisted(Request $request, Response $response): bool
+    {
+        return $response->getStatusCode() >= 300;
```
Comment (damienalexandre) [inline src/Bridge/Symfony/Blacklister/Not2xxBlacklister.php]: This code does not do what the fixer say it does (what about 1xx codes?)

Comment (pyrech) [reply src/Bridge/Symfony/Blacklister/Not2xxBlacklister.php]: Should we really care about 1xx responses? Not sure a PHP app can/should return such statuses code

**Pull request merged**